src/Conversion/ONNXToKrnl/Tensor/Gather.cpp

```diff
@@ -41,18 +41,28 @@ struct ONNXGatherOpLowering : public ConversionPattern {
     Value alloc = insertAllocAndDeallocSimple(
         rewriter, op, outputMemRefType, loc, shapeHelper.dimsForOutput());
 
-    // Save axis and rank info.
+    // Operands and attributes.
+    Value data = operandAdaptor.data();
+    Value indices = operandAdaptor.indices();
     int64_t axisLit = gatherOp.axis();
-    int64_t dataRank = shapeHelper.dataDims.size();
```

Comment:
I have simplified the shapehelper implementation, to make it consistent with other shape helpers.